controllers/common_util_test.go

```diff
@@ -1108,3 +1108,123 @@ func (m *NillCluster) GetAPIReader() client.Reader {
 func (m *NillCluster) Start(ctx context.Context) error {
 	return nil
 }
+
+type dummyClient struct {
```

Comment:
the fakeClient may be an alternative, but possibly not as functional as your dummy.

Reply:
@gtully thanks for the suggestion, I didn't know that fakeClient can intercept calls. I removed the dummyClient in favor of the fakeClient.